pkg/menus/menus.go

```diff
@@ -0,0 +1,162 @@
+package menus
+
+import (
+	"fmt"
+	"log"
+	"os"
+	"strconv"
+
+	ui "github.com/gizak/termui/v3"
+	"github.com/gizak/termui/v3/widgets"
+)
+
+var menuWidth = 50
+
+// Entry contains all the information needed for a boot entry
+type Entry interface {
+	// Label is the string will show in menu
+	Label() string
+	// IsDefault is true means this entry will be hit by default, if there is many default choise, the first on in the list will be choose. the rest will be ignored.
+	IsDefault() bool
+}
+
+// DisplayMenu will present all entries into a menu with numbers, so that user can choose from them.
+func DisplayMenu(menuTitle string, introwords string, location int, entries []Entry) (int, error) {
+	// open a new window
+	if err := ui.Init(); err != nil {
+		log.Printf("failed to initialize termui: %v", err)
+	}
+	// listData contains all choice's labels
+	listData := []string{}
+
+	for i, e := range entries {
+		listData = append(listData, fmt.Sprintf("[%d] %s", i+1, e.Label()))
+	}
+
+	// l is a List item which will present like a menu by termui package
+	l := widgets.NewList()
+	l.Title = menuTitle
+	l.Rows = listData
+	// design the size and location of the menu
+	l.SetRect(0, location, menuWidth, location+len(entries)+2)
+	location += len(entries) + 2
+	l.TextStyle.Fg = ui.ColorWhite
+	ui.Render(l)
+
+	// checkValidFunc is a function for GetInput to check the format of input. Since we want user to choose a option from the menu, the checkValidFunction will check :1.input is a number; 2.input number does not exceed the number of options.
```

Comment:
This comment needs to be fixed. I wonder what editor you are using?

Reply:
I am using vim now, but it is not convenient for me. I wondered will VS code be a good choice?